### ISSUE TITLE
AW Approve Reject and Cancel Route Actions Fail

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#2155]
After sending a form for approval on edge2, any attempt to either approve or reject throws the following error:

```
2020-07-10 11:47:34,190	36252612	ajp-apr-8009-exec-9	ERROR	RESTService	Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.workflow.IWorkflow| name: completeTask
com.tsgrp.opencontent.core.exception.OCRuntimeException: Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.workflow.IWorkflow| name: completeTask
	at com.tsgrp.opencontent.alfresco.core.EmbeddedAlfrescoAdvice.invoke(EmbeddedAlfrescoAdvice.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
	at com.sun.proxy.$Proxy457.completeTask(Unknown Source)
	at com.tsgrp.opencontent.wizard.action.executer.CompleteWizardApprovalRouteActionExecuter.executeImpl(CompleteWizardApprovalRouteActionExecuter.java:36)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.executeImpl(AbstractActionExecuter.java:239)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.execute(AbstractActionExecuter.java:226)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.evaluateAndExecute(AbstractActionExecuter.java:156)
	at com.tsgrp.opencontent.core.action.service.CoreActionService.execute(CoreActionService.java:475)
	at com.tsgrp.opencontent.core.action.service.RESTAction.executeActionPost(RESTAction.java:224)

...
Caused by: org.activiti.engine.ActivitiException: Exception while invoking TaskListener: A signature occurred but no signature event was available in ApprovalDetails.
	at org.activiti.engine.impl.bpmn.helper.ClassDelegate.notify(ClassDelegate.java:123)
	at org.activiti.engine.impl.delegate.TaskListenerInvocation.invoke(TaskListenerInvocation.java:34)
	at org.activiti.engine.impl.delegate.DelegateInvocation.proceed(DelegateInvocation.java:37)
	at org.activiti.engine.impl.delegate.DefaultDelegateInterceptor.handleInvocation(DefaultDelegateInterceptor.java:25)
	at org.activiti.engine.impl.persistence.entity.TaskEntity.fireEvent(TaskEntity.java:740)
	... 80 more
Caused by: com.tsgrp.opencontent.core.exception.OCRuntimeException: A signature occurred but no signature event was available in ApprovalDetails.
	at com.tsgrp.opencontent.wizard.activiti.workflow.listener.CompleteUserTaskListener.notify(CompleteUserTaskListener.java:179)
	at org.activiti.engine.impl.delegate.TaskListenerInvocation.invoke(TaskListenerInvocation.java:34)
```

Also - after cancelling the route (the only way to get the form back to editable), the Form does not go back to the draft state.  The following error is thrown in OC:

```
2020-07-10 11:58:48,816	36927238	ajp-apr-8009-exec-2	DEBUG	CompleteUserTaskListener	Processing Signature Event for role: CR Approval. User: gsteimer. Outcome: approve
2020-07-10 11:58:48,818	36927240	ajp-apr-8009-exec-2	ERROR	RESTService	Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.workflow.IWorkflow| name: completeTask
com.tsgrp.opencontent.core.exception.OCRuntimeException: Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.workflow.IWorkflow| name: completeTask
	at com.tsgrp.opencontent.alfresco.core.EmbeddedAlfrescoAdvice.invoke(EmbeddedAlfrescoAdvice.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
	at com.sun.proxy.$Proxy457.completeTask(Unknown Source)
	at com.tsgrp.opencontent.wizard.action.executer.CompleteWizardApprovalRouteActionExecuter.executeImpl(CompleteWizardApprovalRouteActionExecuter.java:36)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.executeImpl(AbstractActionExecuter.java:239)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.execute(AbstractActionExecuter.java:226)
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.evaluateAndExecute(AbstractActionExecuter.java:156)
	at com.tsgrp.opencontent.core.action.service.CoreActionService.execute(CoreActionService.java:475)
	at com.tsgrp.opencontent.core.action.service.RESTAction.executeActionPost(RESTAction.java:224)
```

Note - I just took a cursory glance, but this looks to be working on edge1.

**Comment by gsteimer** [comment]
After fixing this, take a look to see if this fixes #2152

**Comment by cheinz-tsg** [comment]
Closed with OC Commit #25614. This commit fixed issue #2107. The above issue was an unfortunate side effect of OC Commit #25585 (expanding on workflow statuses) trying to use the ApprovalDetails object in a way that didn't work in an Alfresco-backed OC instance. This prevented the workflow statuses from reaching the "pending" state, which broke most workflow-related actions.
